### PR TITLE
couchdb: support Erlang/OTP-18.0 with patch from upstream

### PR DESCRIPTION
**Pull request by dch**
this is a pre-requisite to committing #41035.

```
$ brew audit --strict couchdb
==> brew style couchdb

1 file inspected, no offenses detected
```

### CODE REVIEW

**Comment by makigumo** [comment]
Why include the whole patch when a link would have sufficed? More something like this.

```ruby
  stable do
    patch do
      # Support Erlang/OTP 18.0 compatibility, see upstream #95cb436
      # It will be in the next CouchDB point release, likely 1.6.2.
      url "https://github.com/apache/couchdb/commit/95cb436.diff"
      sha256 "143114b33f4f134994c4824197e297f1860aa01e8de2c0f7f5e6ca93920a90c6"
    end
  end
```

**Comment by dch** [comment]
@makigumo because the commit is for the git source tree, before `make distcheck` produces the official tarball. The patch has to account for differences over and above the commit. But well spotted ;-)

**Comment by makigumo** [comment]
Ah, I see now, that there is am extra section for ```configure``` at the end. Thanks for the explanation :)

File: Library/Formula/couchdb.rb [hunk 1]
```diff
@@ -32,7 +38,7 @@ def install
     # in the welcome message
     inreplace "etc/couchdb/default.ini.tpl.in" do |s|
       s.gsub! "%package_author_name%", "Homebrew"
-      s.gsub! "%version%", "%version%-1"
+      s.gsub! "%version%", "%version%-3"
```
Comment (mikemcquaid) [inline Library/Formula/couchdb.rb]: Can you use `pkg_version` here instead which gets you the version plus revision?

Comment (dch) [inline Library/Formula/couchdb.rb]: done, produces `version = 1.6.1_3` which seems super reasonable & easy - thanks for the #protip @mikemcquaid

**Comment by dch** [comment]
updated with version = 1.6.1_3 / #{pkg_version} as proposed.

**Comment by dch** [comment]
`brew audit --strict` now passes. yay for CI. pushed again.

**Comment by mikemcquaid** [comment]
Thanks @dch!

**Pull request closed without merging**